miniwin/src/d3drm/backends/sdl3gpu/shaders/src/Common.hlsl

```diff
@@ -18,3 +18,15 @@ struct SceneLight {
     float4 position;
     float4 direction;
 };
+
+#ifdef NO_REGISTER_SPACE
+    #define REGISTER_B0_SPACE1 register(b0)
+    #define REGISTER_B1_SPACE3 register(b0)
+    #define REGISTER_T0_SPACE2 register(t0)
+    #define REGISTER_S0_SPACE2 register(s0)
+#else
+    #define REGISTER_B0_SPACE1 register(b0, space1)
+    #define REGISTER_B1_SPACE3 register(b0, space3)
+    #define REGISTER_T0_SPACE2 register(t0, space2)
+    #define REGISTER_S0_SPACE2 register(s0, space2)
+#endif
```

Comment:
Wouldn't the following be more usable in the future?
```suggestion
#ifdef NO_REGISTER_SPACE
    #define REGISTER_SPACE(REG, SPACE) register(REG)
#else
    #define REGISTER_SPACE(REG, SPACE) register(REG, SPACE)
#endif
```
and use `REGISTER_SPACE(b0, space1)` in the shaders?